jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpConnection.java

```diff
@@ -1523,13 +1534,12 @@ public void succeeded()
                 return;
             }
 
-            // As this is not an upgrade, we can shutdown the output if we know we are not persistent
-            if (_sendCallback._shutdownOut)
-                getEndPoint().shutdownOutput();
+            // Failsafe shutdown (should have been done by SendCallback)
+            if (_sendCallback._shutdownOut && !getEndPoint().isOutputShutdown())
+                IO.close(getEndPoint());
```

Comment:
I don't think this is right. 

Below we even set fill interest to seek EOF, so we cannot close the endpoint here.

Reply:
I'm not sure we need the failsafe at all.  I think the sendCallback is always called.

But I just wanted to be cautious and make sure it is closed.  Maybe a shutdown here would be enough or maybe no failsafe at all.

Reply:
No fail safe.

However, we need to make sure the failure path is correct.